lib/matplotlib/cbook.py

```diff
@@ -361,9 +361,11 @@ class _BoundMethodProxy(object):
     Minor bugfixes by Michael Droettboom
     '''
     def __init__(self, cb):
+        self._hash = hash(cb)
+        self._callbacks = []
         try:
             try:
-                self.inst = ref(cb.im_self)
+                self.inst = ref(cb.im_self, self._destroy)
```

Comment:
Still not crystal clear, I think you ask about the life-cycle of ``cb.im_self``?

In which case, the same life-cycle as if it weren't here, from the definition of a weakref, it doesn't affect the life-cycle of the object it holds, i.e. as soon as the object looses all non-weakref-references it will get destroyed and GC'ed.

```python
from weakref import ref

class Foo(object):
    pass

def make_ref(var):
    print var # 1st print
    r = ref(var)
    print r # 2nd print
    return r

print make_ref(Foo()) # 3rd print
```
prints
```
<__main__.Foo object at 0x7f2600f06650>
<weakref at 0x7f2600ef2e68; to 'Foo' at 0x7f2600f06650>
<weakref at 0x7f2600ef2e68; dead>
```
because var looses scope as soon as the function ends and thus gets destroyed.